### PR TITLE
Dependency injection

### PR DESCRIPTION
**Pull request by OlexTul**
I propose to use this DI option:
1. All classes must have a default constructor.
2. All fields in service classes must be initialized through a constructor with parameters.
3. DTOs with primitive fields or wrapper fields will be created using the default constructor with default values, and they will need to be prepared for use in services.

### CODE REVIEW

**Review by IfLunatic — APPROVED** [review]

**Review by TheDarkRabb1t — APPROVED** [review]

File: src/main/java/executor/service/service/di/Application.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package executor.service.service.di;
```
Comment (jurok3x) [inline src/main/java/executor/service/service/di/Application.java]: Thats perhaps should be executor.service.configuration package

File: src/main/java/executor/service/service/di/ApplicationContext.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package executor.service.service.di;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ApplicationContext {
+
+    private ObjectFactory factory;
+    private final Map<Class<?>, Object> cache = new ConcurrentHashMap<>();
```
Comment (jurok3x) [inline src/main/java/executor/service/service/di/ApplicationContext.java]: why ConcurrentHashMap?

File: src/main/java/executor/service/service/di/Application.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.Map;
+
+public class Application {
+
+    public static ApplicationContext run(String packageToScan, Map<Class, Class> ifc2ImplClass) {
+        JavaConfig config = new JavaConfig(packageToScan, ifc2ImplClass);
+        ApplicationContext context = new ApplicationContext(config);
+        ObjectFactory objectFactory = new ObjectFactory(context);
+        context.setFactory(objectFactory);
```
Comment (jurok3x) [inline src/main/java/executor/service/service/di/Application.java:11]: ObjectFactory depends on context and context depends on ObjectFactory.
It's ok?

File: src/main/java/executor/service/service/di/ApplicationContext.java (hunk 2 above)
Comment (Sokol3783) [reply src/main/java/executor/service/service/di/ApplicationContext.java]: Every web-app work in multi-threaded mode, each new user create new page, if two users or more in same time would do the same actions, it should be two or more object, but in HashMap or LinkedHashMap would stay only class which had been called last. So we should use synchronize or concurrent.

File: src/main/java/executor/service/service/di/InjectByTypeAnnotationObjectConfigurator.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                && fieldType != Boolean.class
+                && fieldType != Character.class
+                && fieldType != Byte.class
+                && fieldType != Short.class
+                && fieldType != Integer.class
+                && fieldType != Long.class
+                && fieldType != Float.class
+                && fieldType != Double.class
+                && fieldType != Void.class;
+    }
```
Comment (jurok3x) [inline src/main/java/executor/service/service/di/InjectByTypeAnnotationObjectConfigurator.java]: `
or 
` return !fieldType.isPrimitive() && !fieldType.getName().startsWith(excludedPackage);`
```suggestion
private boolean isFieldNotPrimitiveOrWrapper(Field field) {
        Class<?> fieldType = field.getType();
    List<Class<?>> excludedClasses = Arrays.asList(
        String.class, Boolean.class, Character.class,
        Byte.class, Short.class, Integer.class,
  Long.class, Float.class, Double.class, Void.class
);
return !fieldType.isPrimitive() && !excludedClasses.contains(fieldType);
//or
private boolean isFieldNotPrimitiveOrWrapper(Field field) {
        Class<?> fieldType = field.getType();
String excludedPackage = "java.lang";
return !fieldType.isPrimitive() && !excludedClasses.contains(fieldType);
```

**Review by jurok3x — DISMISSED** [review]

File: src/main/java/executor/service/service/di/Application.java (hunk 1 above)
Comment (OlexTul) [reply src/main/java/executor/service/service/di/Application.java]: I changed the package.

File: src/main/java/executor/service/service/di/InjectByTypeAnnotationObjectConfigurator.java (hunk 4 above)
Comment (OlexTul) [reply src/main/java/executor/service/service/di/InjectByTypeAnnotationObjectConfigurator.java:38]: I don't mind, we can replace

**Pull request merged**